### PR TITLE
Conventions for Cargo.toml files.

### PR DESCRIPTION
**Pull request by joshtriplett**

### CODE REVIEW

File: guide/guide.md [hunk 1]
```diff
@@ -225,3 +225,78 @@ if y {
...
+key-value pairs in that section, or between key-value pairs in a section.
+
+Sort key names alphabetically within each section, with the exception of the
+`[package]` section. Put the `[package]` section at the top of the file; put
+the `name` and `version` keys in that order at the top of that section,
+followed by the remaining keys other than `description` in alphabetical order,
+followed by the `description` at the end of that section.
+
+Don't use quotes around any standard key names; use bare keys. Only use quoted
+keys for non-standard keys whose names require them, and avoid introducing such
```
Comment (killercup) [inline guide/guide.md]: > whose names require them

Can you specify when that is required or link to TOML spec? Actually: Link to the TOML spec anyway :)

**Comment by joshtriplett** [comment]
@killercup I *think* that Cargo follows the current version, <https://github.com/toml-lang/toml/blob/master/versions/en/toml-v0.4.0.md>. Resident Cargo expert @alexcrichton would know for sure.

File: guide/guide.md (hunk 1 above)
Comment (joshtriplett) [inline guide/guide.md]: Quoting <https://github.com/toml-lang/toml/blob/master/versions/en/toml-v0.4.0.md#table>:
> Keys may be either bare or quoted. Bare keys may only contain letters, numbers, underscores, and dashes (`A-Za-z0-9_-`). Quoted keys follow the exact same rules as basic strings and allow you to use a much broader set of key names. Best practice is to use bare keys except when absolutely necessary.

**Comment by alexcrichton** [comment]
Ah yes Cargo uses toml-rs which follows the TOML spec as closely as it can, so the master version of the TOML spec is what it follows. I forget if that's *exactly* 0.4.0 or slightly beyond it wrt a few small cases.

**Comment by joshtriplett** [comment]
@alexcrichton Would it make sense, then, to link to the latest version of the spec and update that reference if Cargo/toml-rs ever needs to refer to an older snapshot rather than the latest version?

**Comment by alexcrichton** [comment]
Sure!

**Comment by nrc** [comment]
This LGTM, and I think is ready for FCP

File: guide/guide.md [hunk 2]
```diff
@@ -225,3 +225,78 @@ if y {
...
+version = "4.5.6"
+```
+
+### Metadata conventions
+
+The authors list should consist of strings that each contain an author name
+followed by an email address in angle brackets: `Full Name <email@address>`.
+It should not contain bare email addresses, or names without email addresses.
+(The authors list may also include a mailing list address without an associated
+name.)
```
Comment (nrc) [inline guide/guide.md]: Nit: I think the full stop goes outside the parens, here and below. I bet this is something different style guides disagree on though.

File: guide/guide.md [hunk 3]
```diff
@@ -225,3 +225,78 @@ if y {
...
+[dependencies.extremely_long_crate_name_goes_here]
+path = "extremely_long_path_name_goes_right_here"
+version = "4.5.6"
+```
+
+### Metadata conventions
+
+The authors list should consist of strings that each contain an author name
+followed by an email address in angle brackets: `Full Name <email@address>`.
+It should not contain bare email addresses, or names without email addresses.
```
Comment (nrc) [inline guide/guide.md]: Is there a case where people might want credit without giving contact info?

File: guide/guide.md [hunk 4]
```diff
@@ -225,3 +225,78 @@ if y {
...
+(The authors list may also include a mailing list address without an associated
+name.)
+
+The license field should contain a valid SPDX expression, using valid SPDX
```
Comment (nrc) [inline guide/guide.md]: Could you link to what SPDX is?

File: guide/guide.md [hunk 5]
```diff
@@ -225,3 +225,78 @@ if y {
...
+license names. (As an exception, by widespread convention, the license field
+may use `/` in place of ` OR `; for example, `MIT/Apache-2.0`.)
+
+The homepage field, if present, should consist of a single URL, including the
+scheme (e.g. `https://example.org/`, not just `example.org`.)
+
+Within the description field, wrap text at 78 columns. Don't start the
```
Comment (nrc) [inline guide/guide.md]: Is this 80 cols minus the start and end quote marks? Or 78 including the quote marks?

File: guide/guide.md (hunk 2 above)
Comment (solson) [inline guide/guide.md]: @nrc The rule I've heard is that when part of sentence, the full stop is outside (like this). (But when the parens enclose an entire sentence, it's inside, like this.)

File: guide/guide.md (hunk 3 above)
Comment (joshtriplett) [inline guide/guide.md]: That exact case led me to write "should" instead of "must" here.

File: guide/guide.md (hunk 2 above)
Comment (joshtriplett) [inline guide/guide.md]: I've heard the same rule as @solson.  In this case, the parenthetical contains a sentence; the sentence doesn't contain a parenthetical.

File: guide/guide.md (hunk 4 above)
Comment (joshtriplett) [inline guide/guide.md]: Will do.

File: guide/guide.md (hunk 5 above)
Comment (joshtriplett) [inline guide/guide.md]: I intended this as 78 columns of description text, not counting the quote marks.

**Comment by nrc** [comment]
This has been in FCP for about a month, and module  some haggling about grammar nits has no significant comments, so I'm merging it.

**Pull request merged**